### DISCUSSION TITLE
how to set link attribute when writing mdf file

### DISCUSSION THREAD

**Discussion by hljjj** [danielhrisca/asammdf#794] [Q&A]
Hi Daniel,

I am using asammdf to read and wirte mdf file, and I have some problems,here is my code:

```python
    mdf = MDF(origin_mdf_path)
    new_mdf = MDF()

    channels = []
    for m in mdf.iter_channels():
        if m.name != 'VideoRawdata_VC0':
            channels.append(m)
        else:
            new_samples = rebuild_samples_async(m.samples, logger, shape, config_path, so_path)
            new_signal = Signal(
                samples=new_samples,
                timestamps=m.timestamps,
                unit=m.unit,
                name=m.name,
                conversion=m.conversion,
                raw=m.raw,
                master_metadata=m.master_metadata,
                display_names=m.display_names,
                attachment=m.attachment,
                stream_sync=m.stream_sync,
                invalidation_bits=m.invalidation_bits,
                source=m.source,
                encoding=m.encoding,
                group_index=m.group_index,
                channel_index=m.channel_index)
            channels.append(new_signal)
            
    new_mdf.append(channels)
    new_mdf.save(new_mdf_path)
```

In the origin mdf file, the channel 'VideoRawdata_VC0' has a attribute name 'cn_data', it's a link to signal data block, like this, the address of the signal data block is '0x19D0B448'
![the original cn_data](https://user-images.githubusercontent.com/45890184/205602831-a924e6b2-2432-479a-8697-9c81923cca9f.png)

in the new mdf file, the value of 'cn_data' becomes  ‘0x0000’, the correct address of the signal data should be ‘0x19D09E28’
![the new cn_data](https://user-images.githubusercontent.com/45890184/205604303-728c0a2b-af12-4c24-9de6-5f7d0d3f1e4f.png)
![the address of datalist](https://user-images.githubusercontent.com/45890184/205604704-194efa5b-ba9b-49f7-a24b-5a6712f0f084.png)

Are there any way to keep the cn_data link to the datalist in the new mdf file?
How can I get this attribute in asammdf? 

Thanks

**Comment by danielhrisca** [comment]
In the new file, the channel samples are not stored in signal data blocks (the linked address) but instead directly in the data blocks of the data group.

**Reply by hljjj** [reply]
Hi, Daniel, thank you for your reply. 
I still have some question, I am new to mdf,  I use CANape to playback vedio stored in the mdf file，it can not recognize the  new mdf file(it's no problem when I use python code to read)， I think it may due to the signal data did not  stored in the correct path(actually I'm not sure)，how can I stored channel sample in the signal data blocks instead of the data group? or are there any other library support this function?
